src/libexpr/primops/fetchGit.cc

```diff
@@ -34,12 +36,18 @@ static void prim_fetchGit(EvalState & state, const Pos & pos, Value * * args, Va
                 name = state.forceStringNoCtx(*attr.value, *attr.pos);
             else if (n == "submodules")
                 fetchSubmodules = state.forceBool(*attr.value, *attr.pos);
+            else if (n == "narHash")
+                narHash = state.forceStringNoCtx(*attr.value, *attr.pos);
             else
                 throw EvalError("unsupported argument '%s' to 'fetchGit', at %s", attr.name, *attr.pos);
         }
 
         if (url.empty())
             throw EvalError(format("'url' argument required, at %1%") % pos);
+        if (narHash && !rev)
+            throw EvalError(
+                format("'narHash' can only be used for 'fetchGit' if 'rev' is set, at %1%") % pos
```

Comment:
I think this is too strict. If you use a tag ref, it's fine to specify a `narHash` but not a `rev`.

Reply:
Agreed. Just realized that I allowed both revs and refs in the original PR.